### ISSUE TITLE
According to my research second-digit tests are more reliable for detecting election fraud

### ISSUE THREAD

**Issue by draeder** [cjph8914/2020_benfords#12]
See: http://www-personal.umich.edu/~wmebane/pm06.ps, https://www.degruyter.com/view/journals/jbnst/231/5-6/article-p719.xml and https://www.researchgate.net/publication/275305550_Comment_on_Benford's_Law_and_the_Detection_of_Election_Fraud

**Comment by ghost** [comment]
First-digit tests were also found in this thread to give misleading results: https://github.com/cjph8914/2020_benfords/issues/5

Accordingly, I suggest that we **immediately** implement a second-digit test. 

(minor edit)

**Comment by dshield55** [comment]
How do you do a 2nd digit test?

I just did one off a large data set not suspected of fraud. Excel. When I do the 1st digit test you can see the curve so well. When I do the 2nd digit test, it's not near as dramatic even though there's more 0s than 1s than 2s.

Isn't there a d number or something that is calculated to determine how good a fit the curve is? Anyone know how to do that?

**Comment by ghost** [comment]
According to Benford's law, the 2nd digit will likely have a different distribution than the 1st digit. Maybe draeder (or his papers, linked above) has the exact formula to use.

**Comment by ghost** [comment]
I recovered the second-digit factors from this paper: https://www.jstor.org/stable/23011436

0 - 0.120
1 - 0.114
2 - 0.109
3 - 0.104
4 - 0.100
5 - 0.097
6 - 0.093
7 - 0.090
8 - 0.088
9 - 0.085

Mean: 4.187

Academic literature suggests all kinds of pitfalls when it comes to interpreting analyses like this, but for now let's not overcomplicate the issue. Just do a second-figure test, and see if it agrees with the above, and if you know how also calculate p-values.

Note that the analysis is based on a pretty implausible hypothesis, namely that the _totals_ for each precinct are taken out of thin air.

**Comment by dshield55** [comment]
I did just a basic graph all the 2nd digits operation here, no other fancy analysis. The bottom one isn't expected to be fraudulent and you can see it still follows the rules of more 1s than 2s than 3s, etc. However with Biden in Chicago, it's jacked on the 2nd digits.

I understand the reasons we can't necessarily trust the 1st digits, but is there a reason we wouldn't be able to trust the distribution of 2nd digits?

<img width="502" alt="2nddigitexperiment" src="https://user-images.githubusercontent.com/5439391/98445676-1ad21d80-20d6-11eb-84ca-849de379de76.png">

**Comment by ghost** [comment]
> I understand the reasons we can't necessarily trust the 1st digits, but is there a reason we wouldn't be able to trust the distribution of 2nd digits?


Academic literature suggests that one needs to be careful about drawing conclusions, see e.g.  https://doi.org/10.1515/jbnst-2011-5-610 - Anyhow, keep these diagrams coming!

(edit: removed diagrams from quote)

**Comment by dshield55** [comment]
Here's the percentages for the my last posted chart alongside the second-digit factors. The Trump one matches quite well, but not Biden.
<img width="185" alt="whackbenfordsec" src="https://user-images.githubusercontent.com/5439391/98445939-d051a080-20d7-11eb-9b4e-a2eb23a37c5f.png">

**Comment by dshield55** [comment]
*I'm out of time and likely won't be contributing more

In Chicago, Trump appears to hold up with 2nd digits but Biden does not. No fancy analysis here, just eyeballing it.

With a second digit analysis, I don't know what to do with entries that don't have a second digit, so I counted 119 of them as null, was I supposed to add them to the zeroes? The lowest biden got in a precinct was 33, but Trump got somes 7s and 9s, etc.


<img width="661" alt="chicago2nd" src="https://user-images.githubusercontent.com/5439391/98446707-88814800-20dc-11eb-885e-56ce113a267b.png">

**Comment by ghost** [comment]
> In Chicago, Trump appears to hold up with 2nd digits but Biden does not. No fancy analysis here, just eyeballing it.

I have tried to outline how the second digits of results in cities could just as well follow a uniform distribution, here: https://github.com/cjph8914/2020_benfords/issues/16

I don't understand, however, why the results of each candidate don't follow the same pattern. Maybe it has to do with how precinct sizes are planned in inner cities vs. in suburban areas, but that's just a hypothesis.

**Comment by chavenor** [comment]
Yes, there are wards voting like 98% one way.  So this would show up in the graph would it not?  https://results.enr.clarityelections.com/PA/Allegheny/63905/Web02.193333/#/cid/0104

**Comment by testes-t** [comment]
If there are many wards voting 90-98% for one candidate, that's also not random. Voters seem to have introduced bias to the numbers by favouring one of the candidates.

* They have to count each vote starting from zero, so Benford's Law should apply, but:
* There are several other correlations and contingencies that could mess up the pure Benford pattern.

**Comment by PoiSonPeZ** [comment]
Did you guys ever consider that you are combining two separate elections - mail-in ballots and in-person ballots? They are being collected and counted separately. The mail in ballots have shown a considerable Democrat preference. It's almost as if two separate elections are occurring. I suspect this is the reason for the discrepancy you are seeing.

**Comment by ghost** [comment]
> In Chicago, Trump appears to hold up with 2nd digits but Biden does not. No fancy analysis here, just eyeballing it.

> <img alt="chicago2nd" width="661" src="https://user-images.githubusercontent.com/5439391/98446707-88814800-20dc-11eb-885e-56ce113a267b.png">

I did the same analysis, and my diagrams deviate slightly from yours (e.g. I have 210 zeros for Biden, but otherwise it's more or less the same). I'm not sure exactly why we see the pattern in Trump's second digits and not in Biden's.
